### ISSUE TITLE
Automation for setting milestones on PRs and issues is broken

### ISSUE THREAD

**Issue by nickboldt** [eclipse/che#22120]
### Is your task related to a problem? Please describe

The GH action https://github.com/benoitf/che-sync-remote/actions/workflows/sync.yml used to set milestones on PRs and issues has broken.

We need to look at WHY and then resurrect it; or we could consider moving to a simpler python script that uses the [GH API](https://github.com/redhat-developer/devspaces/blob/devspaces-3-rhel-8/product/ghira/ghira#L15) to manipulate issues and pull requests. 

### Describe the solution you'd like

Fix the existing TS app or move to python script.

Move the script to a centalized place, like https://github.com/eclipse-che/che-release/

**Comment by nickboldt** [comment]
This might be P1 for the next sprint, depending on opinions from @ibuziuk and @l0rd  about how it impacts data workflows for documentation & the Thursday PR/issue triage calls. cc: @max-cx as well.

**Comment by l0rd** [comment]
The repo has been [moved in che-incubator](https://github.com/che-incubator/che-sync-remote).

**Comment by nickboldt** [comment]
Related issue: https://issues.redhat.com/browse/CRW-4349

**Comment by ibuziuk** [comment]
@nickboldt can we prio this issue for the next sprint?

**Comment by nickboldt** [comment]
PR: https://github.com/che-incubator/che-sync-remote/pull/26

```
Update milestone with params {
  owner: 'eclipse-che',
  repo: 'che-theia',
  milestone_number: 71,
  title: '7.64',
  state: 'closed'
}
RequestError [HttpError]: Repository was archived so is read-only.
```

Looks like we need to remove archived products from https://raw.githubusercontent.com/eclipse/che/master/.repositories.yaml ...

**Comment by nickboldt** [comment]
milestones added to these repos:

```
eclipse/che
7.74
7.75

eclipse-che/che-server
8.x
7.75
7.74

che-incubator/chectl
8.x
7.x (Later, Future, Never)
7.75
7.74

che-incubator/che-code
8.x
7.75
7.74

eclipse-che/che-dashboard
8.x
7.x (Later, Future, Never)
7.75
7.74

eclipse-che/che-devfile-registry
8.x
7.x (Later, Future, Never)
7.75
7.74

eclipse-che/che-docs
8.x
7.x (Later, Future, Never)
7.75
7.74

eclipse-che/che-machine-exec
8.x
7.x (Later, Future, Never)
7.75
7.74

eclipse-che/che-operator
7.16.0
8.x
7.x (Later, Future, Never)
7.75
7.74

eclipse-che/che-plugin-registry
8.x
7.x (Later, Future, Never)
7.75
7.74
```

Repos without milestones:

```
eclipse-che/che-website
eclipse-che/che-workspace-client
che-incubator/configbump
che-incubator/workspace-data-sync
che-incubator/che-workspace-telemetry-client
che-incubator/kubernetes-image-puller
eclipse-che/blog
eclipse-che/che-docs-vale-style
eclipse-che/che-release
che-incubator/che-deploy-action
che-incubator/devfile-converter
che-incubator/happy-path-tests-action
che-incubator/header-rewrite-traefik-plugin
che-incubator/jetbrains-editor-images
che-incubator/kubernetes-image-puller-operator
devfile/registry
devfile/developer-images
devfile/devworkspace-operator-docs
che-incubator/check-license-header
che-incubator/devworkspace-telemetry-woopra-plugin
che-incubator/setup-minikube-action
che-incubator/dependencies-license-action
che-incubator/dash-licenses
che-incubator/devfile-api
```

Also:

```
devfile/devworkspace-operator
v0.19.x
v0.20.x
v0.21.x
v0.23.x
v0.22.x
```

**Comment by nickboldt** [comment]
```
Query String = repo:eclipse/che repo:eclipse-che/che-server/ repo:che-incubator/chectl repo:che-incubator/che-code repo:eclipse-che/che-dashboard repo:eclipse-che/che-devfile-registry repo:eclipse-che/che-docs repo:eclipse-che/che-machine-exec repo:eclipse-che/che-operator repo:eclipse-che/che-plugin-registry is:pr merged:>=2023-09-11T14:05:03.485Z is:merged no:milestone
```

Not sure yet if we need to add more repos to the list... and if so, how?

**Comment by l0rd** [comment]
Do we really need to maintain a bot that sets milestones of merged PRs? 

When a merged PR belongs to a git tag then GitHub already shows the label:

![image](https://github.com/eclipse/che/assets/606959/cc3f0154-3a56-407a-994d-12ed15e254cc) ([link](https://github.com/devfile/devworkspace-operator/pull/1138))

![image](https://github.com/eclipse/che/assets/606959/8111af08-d2b9-4b5a-8a68-d0c428b64e16) ([link](https://github.com/eclipse-che/che-server/pull/499))

**Comment by nickboldt** [comment]
I thought the point of this was to set milestones (before a release), not tags (after a release) ... but if we no longer need this to facilitate triage and creation of N&N (after a release), then I'm happy to sunset this tool.

**Comment by l0rd** [comment]
> I thought the point of this was to set milestones (before a release), not tags (after a release)

The reasoning is: if there is no tag, then the PR will be included in next release. And if the tag is there, then it matches the release that includes the PR.

**Comment by nickboldt** [comment]
Good point. So we don't need milestones for che project anymore?

**Comment by ibuziuk** [comment]
During the retrospective we had the discussion about this issue, and identified it as an action item for Dev Tools week.
Basically, the problem is that it is very hard at the moment to identify which issues were closed in particular Eclipse Che version. Previously we used milestones [1] for that which are not accurate e.g. 7.72.0 and 7.73.0 have only 3 issues. My understanding is that we should either set milestones manually on each issue before closing, or bring back the automation. 

[1] https://github.com/eclipse/che/milestones?state=closed
